.github/workflows/build.yml

```diff
@@ -206,9 +206,14 @@ jobs:
 
   check:
     name: Check all builds successful
+    if: ${{ !cancelled() }}
     runs-on: ubuntu-latest
     needs: [push-ghcr]
     steps:
+      - name: Exit on failure
+        if: ${{ needs.push-ghcr.result == 'failure' }}
```

Comment:
This is very nice!  Might have to ~~steal~~ reference this for my own workflows...